### ISSUE TITLE
Create `attr` API like `prop`, but attributes

### ISSUE THREAD

**Issue by treshugart** [skatejs/skatejs#818]
The code to do this (as I've wanted to do this within a project) would be something like:

``` js
const attr = Object.keys(prop).reduce((prev, curr) => {
  prev[curr] = prop[curr].bind(null, { attribute: true });
  return prev;
}, {});
```

@bradleyayers had an idea to try and completely automate this behaviour, but when discussed, we found a few drawbacks:
- If you _always_ set linked properties, how do you:
  - link non-string data; you can do this with serialisation but there's perf implications there
  - performance of setting attributes hits 3-times as hard as setting properties
- You could just not keep the attribute in sync, and
  - if the attribute is set, reflect it to the property
  - if the property is  set, do _not_ reflect it to the attribute
    - the property becomes the source of truth
    - but what happens if there's styling based on the attr value changing and the prop was set?
  - native DOM seems to behave inconsistently in that linked attrs and props aren't always reflected
    - once they become out of sync (some combination of setting prop or attr separately) they never reflect values back to one another; i.e. `input[checked]` (this is actually maddening behaviour when debugging things, so I'm not sure it's good we follow the DOM here)

**Comment by treshugart** [comment]
If we do #838 then we can probably turn on reflection by default for all props except for the slot prop in #863.

**Comment by andreawyss** [comment]
Now that we solved #838 we can close this one. There is no longer a need for it.
